pallets/duster/src/lib.rs

```diff
@@ -297,34 +254,33 @@ use orml_traits::currency::OnDust;
 use sp_std::marker::PhantomData;
 pub struct DusterWhitelist<T>(PhantomData<T>);
 
-impl<T: Config> OnDust<T::AccountId, T::CurrencyId, T::Balance> for Pallet<T> {
-	fn on_dust(who: &T::AccountId, currency_id: T::CurrencyId, amount: T::Balance) {
-		if let Some(dust_dest_account) = Self::dust_dest_account() {
-			let _ = Self::transfer_dust(who, &dust_dest_account, currency_id, amount);
-		}
+impl<T: Config> OnDust<T::AccountId, T::AssetId, Balance> for Pallet<T> {
+	fn on_dust(who: &T::AccountId, currency_id: T::AssetId, amount: Balance) {
+		let _ = Self::transfer_dust(who, &T::TreasuryAccountId::get(), currency_id, amount);
 	}
 }
 
 impl<T: Config> Contains<T::AccountId> for DusterWhitelist<T> {
 	fn contains(t: &T::AccountId) -> bool {
-		AccountBlacklist::<T>::contains_key(t)
+		AccountWhitelist::<T>::contains_key(t)
 	}
 }
 
 use frame_support::sp_runtime::DispatchError;
+use frame_support::traits::tokens::Preservation;
 use hydradx_traits::pools::DustRemovalAccountWhitelist;
 
 impl<T: Config> DustRemovalAccountWhitelist<T::AccountId> for Pallet<T> {
 	type Error = DispatchError;
 
 	fn add_account(account: &T::AccountId) -> Result<(), Self::Error> {
-		AccountBlacklist::<T>::insert(account, ());
+		AccountWhitelist::<T>::insert(account, ());
 		Ok(())
 	}
 
 	fn remove_account(account: &T::AccountId) -> Result<(), Self::Error> {
-		AccountBlacklist::<T>::mutate(account, |maybe_account| -> Result<(), DispatchError> {
-			ensure!(!maybe_account.is_none(), Error::<T>::AccountNotBlacklisted);
+		AccountWhitelist::<T>::mutate(account, |maybe_account| -> Result<(), DispatchError> {
```

Comment:
we should be just use remove method on the storage. 

simpler.

Reply:
i think by using remove, we would change the existing behaviour, because this mutate checks is the account is already whitelisted, if not it returns error AccountNotWhitelisted, otherwise removes it from whitelist..